subsys/bluetooth/shell/has.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file
+ * @brief Bluetooth Hearing Access Service (HAS) shell.
+ *
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/audio/has.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "bt.h"
+
+static int cmd_preset_reg(const struct shell *sh, size_t argc, char **argv)
+{
+	int err;
+	struct bt_has_preset_register_param param = {
+		.index = strtol(argv[1], NULL, 0),
+		.properties = strtol(argv[2], NULL, 0),
```

Comment:
We are missing parse error handling. We have a shiny new API wrapping `strtol` that has an error output.

```suggestion
		.index = shell_strtol(argv[1], 0, &parse_err_index),
		.properties = shell_strtol(argv[2], 0, &parse_err_properties),
```

Reply:
I forgot it's also possible to use the same `err`-variable for both calls. It will only be changed when there is an error. (It has to be initialized to 0 before the calls then.)